src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -196,13 +231,36 @@ void RuntimeConfigurator::update_loop_info(const lowered::LinearIRCPtr& linear_i
             expanded_loop_info->update_ptr_increments(ptr_increments);
         }
         expanded_loop_info->update_finalization_offsets(updated_finalization_offsets);
+    };
+
+    update_loop_info = [&](const lowered::LoopInfoPtr& loop_info) {
+        if (const auto unified_loop_info = ov::as_type_ptr<lowered::UnifiedLoopInfo>(loop_info)) {
+            update_unified_loop_info(unified_loop_info);
+        } else if (const auto expanded_loop_info = ov::as_type_ptr<lowered::ExpandedLoopInfo>(loop_info)) {
+            update_expanded_loop_info(expanded_loop_info);
+        } else {
+            OPENVINO_THROW("Failed to update loop info: unknown type!");
+        }
+    };
+
+    const auto& loop_map = linear_ir->get_loop_manager()->get_map();
+    for (const auto& p : loop_map) {
+        const auto& expanded_loop_info = ov::as_type_ptr<lowered::ExpandedLoopInfo>(p.second);
+        OPENVINO_ASSERT(expanded_loop_info, "UpdateLoopInfo expects ExpandedLoopInfo in LoopManager");
```

Comment:
It's just remainders after refactoring.
I'd say we can even skip validation because it's done before in `ValidateExpandedLoops` pass in control flow.